src/net.cpp

```diff
@@ -556,7 +556,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
                              pszDest ? pszDest : "",
                              conn_type,
                              /*inbound_onion=*/false,
-                             std::move(i2p_transient_session));
+                             CNodeOptions{ .i2p_sam_session = std::move(i2p_transient_session) });
```

Comment:
In commit "net: add CNodeOptions for optional CNode constructor params" (9dccc3328eeaf9cd66518d812c878def5d014e36)

I'd shorten this as follows, but it's also reasonable to keep if you think the struct name is helpful.

```diff
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -556,7 +556,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
                              pszDest ? pszDest : "",
                              conn_type,
                              /*inbound_onion=*/false,
-                             CNodeOptions{ .i2p_sam_session = std::move(i2p_transient_session) });
+                             { .i2p_sam_session = std::move(i2p_transient_session) });
     pnode->AddRef();
 
     // We're making a new connection, harvest entropy from the time (and our peer count)

```

The short version is nice because it's basically a substitute for having named arguments in c++.

Same comment also applies to later commits, as well.

Reply:
Agree, but I don't think this is possible until we drop support for gcc 8? https://github.com/bitcoin/bitcoin/pull/25962#discussion_r960500526